src/App.tsx

```diff
@@ -1,24 +1,212 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from './types/Todo';
+import { FILTERS } from './types/FILTERS';
+import * as todoService from './api/todos';
+import { USER_ID } from './utils/constants';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { TodoList } from './components/TodoList';
+import { Filter } from './components/Filter';
+import { NewTodo } from './components/NewTodo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [filter, setFilter] = useState(FILTERS.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [loading, setLoading] = useState<number[]>([]);
+
+  const addTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setErrorMessage('');
+
+    if (query.trim() === '') {
+      setErrorMessage('Title can\'t be empty');
+      setQuery('');
+
+      return;
+    }
+
+    setIsAdding(true);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: query.trim(),
+      completed: false,
+    });
+    todoService.createTodo({
+      userId: USER_ID,
+      title: query.trim(),
+      completed: false,
+    })
+      .then(newTodo => {
+        setQuery('');
+        setTodos((currentTodos: Todo[]) => [...currentTodos, newTodo]);
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+
+        throw error;
```

Comment:
do not throw it - it will case page crash
you already handle it and show message